flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/LocalExecutor.java

```diff
@@ -462,7 +462,7 @@ public ResolvedExpression parseSqlExpression(String sqlExpression, TableSchema i
 	@Override
 	public ResultDescriptor executeQuery(String sessionId, String query) throws SqlExecutionException {
 		final ExecutionContext<?> context = getExecutionContext(sessionId);
-		return executeQueryInternal(sessionId, context, query);
+		return context.wrapClassLoader(() -> executeQueryInternal(sessionId, context, query));
```

Comment:
most operations in `executeQueryInternal ` method and `executeUpdateInternal` method are already wrapped in user classloader, only `deployer.deploy()` is  needed. Otherwise, it's better we should remove those wrappers.  btw, add some tests in sql client to verify the fix ?